File: c++/p003.cpp
#include <iostream>
#include <cmath>

using namespace std;

int smallestFactor(long x) {
	for (int i = 2; i <= floor(pow(x, 0.5)); i++) {
		if (x % i == 0) {
			return i;
		};
	};
	return x;  
};

int main() {
	long x = 600851475143;
	while (true) {
		int y = smallestFactor(x);
		if (y < x) {
			x /= y;
		} else {
			cout << "Result: " << x;
			return x;
		}
	};
	cout << "Result: " << x;
	
};

File: ruby/p003.rb
def smallestFactor(x) 
	for i in 2..((x**0.5).floor()) do
		if x % i == 0 then
			return i
		end
	end
	return x  
end

def main() 
	x = 600851475143
	while true do
		y = smallestFactor(x)
		if y < x then
			x = x / y
		else 
			return x
		end
	end
end
puts main()